api/proxy/store/eigenda_manager.go

```diff
@@ -91,18 +92,115 @@ func (m *EigenDAManager) Get(ctx context.Context,
+	case certs.V0VersionByte:
+		if m.eigenda == nil {
+			return nil, errors.New("received CertV0 but EigenDA V1 client is not initialized")
+		}
+		return m.getEigenDAV1(ctx, versionedCert, opts)
+	case certs.V1VersionByte, certs.V2VersionByte:
+		if m.eigendaV2 == nil {
+			return nil, errors.New("received EigenDAV2 cert but EigenDA V2 client is not initialized")
+		}
+		return m.getEigenDAV2(ctx, versionedCert, opts)
+	default:
+		return nil, fmt.Errorf("cert version unknown: %b", versionedCert.Version)
+	}
+}
+
+// getEigenDAV1 will attempt to retrieve a blob for the given versionedCert
+// from cache, EigenDA V1, and fallback storage.
+// TODO: we should also add the v1 RetrievalClient to retrieve from the validators directly
+// in case the v1 disperser is down, the same way we do for v2.
+func (m *EigenDAManager) getEigenDAV1(
+	ctx context.Context,
+	versionedCert certs.VersionedCert,
+	opts common.GETOpts,
+) ([]byte, error) {
+	if opts.ReturnEncodedPayload {
+		return nil, fmt.Errorf("returning encoded payload is not supported for V0 certificates (EigenDA V1)")
 	}
-	if versionedCert.Version == certs.V1VersionByte && m.eigendaV2 == nil {
-		return nil, errors.New("expected EigenDA V2 backend for DA commitment type with CertV1")
+
+	verifyFnForSecondary := func(ctx context.Context, cert []byte, payload []byte, l1InclusionBlockNumber uint64) error {
+		// We don't add the cert version because EigenDA V1 only supports [certs.V0VersionByte] Certs.
+		// We also don't use the l1InclusionBlockNumber because Recency check is only supported by EigenDA V2.
+		// TODO: we should decouple the Verify function into a VerifyCert and VerifyBlob function,
+		// and only verify the cert once here, before retrievals, and then only verify the blob commitment
+		// after retrievals.
+		return m.eigenda.Verify(ctx, cert, payload)
 	}
 
-	// The eigendav2 verify function verifies the cert, but not the kzg commitment.
-	// TODO: should we be verifying those when retrieving from secondary storages like S3...?
-	verifyMethod, err := m.getVerifyMethod(versionedCert.Version)
+	var readErrors []error
+	// 1 - read payload from cache if enabled
+	// Secondary storages (cache and fallback) store payloads instead of blobs.
+	// For simplicity, we bypass secondary storages when requesting encoded payloads,
+	// since those requests are only for secure integrations and run by provers/challengers.
+	// TODO: would be nice to store blobs instead of payloads in secondary storages, such that we could standardize all
+	// storages and make them all implement the [clients.PayloadRetriever] interface.
+	// We could then get rid of the proxy notion of caches/fallbacks and only have storages.
+	if m.secondary.CachingEnabled() && !opts.ReturnEncodedPayload {
+		m.log.Debug("Retrieving payload from cached backends")
+		payload, err := m.secondary.MultiSourceRead(ctx,
+			versionedCert.SerializedCert, false, verifyFnForSecondary, opts.L1InclusionBlockNum)
```

Comment:
It's a bit misleading to pass in `opts.L1InclusionBlockNum` here (and below), given you've stated in the doc above that this isn't a relevant concept for v1.

Would it make sense to pass in a 0 or something instead, to make it clear what's actually being checked here?